matchzoo/datapack.py

```diff
@@ -0,0 +1,115 @@
+"""Matchzoo DataPack, pair-wise tuple (feature) and context as input."""
+
+import typing
+from pathlib import Path
+
+import dill
+import pandas as pd
+
+
+class DataPack(object):
+    """
+    Matchzoo DataPack data structure, store dataframe and context.
+
+    Example:
+        >>> features = [([1,3], [2,3]), ([3,0], [1,6])]
+        >>> context = {'vocab_size': 2000}
+        >>> dp = DataPack(data=features,
+        ...               context=context)
+        >>> type(dp.sample(1))
+        <class 'matchzoo.datapack.DataPack'>
+        >>> len(dp)
+        2
+        >>> features, context = dp.dataframe, dp.context
+        >>> context
+        {'vocab_size': 2000}
+    """
+
+    DATA_FILENAME = 'data.dill'
+
+    def __init__(self,
+                 data: list,
+                 context: dict={}):
+        """Initialize."""
+        self._dataframe = pd.DataFrame(data)
+        self._context = context
+
+    def __len__(self) -> int:
+        """Get size of the data pack."""
```

Comment:
"size" means number of elements in pandas, here you mean the number of samples, or the number of rows.